centralstore/redis_store.go

```diff
@@ -742,28 +701,26 @@ func newTraceStateProcessor(cfg traceStateProcessorConfig, clock clockwork.Clock
 		},
 		config: cfg,
 		clock:  clock,
+		done:   make(chan struct{}),
 		tracer: tracer,
 	}
 
 	return s
 }
 
-func (t *traceStateProcessor) Start(ctx context.Context, redis redis.Client) error {
+func (t *traceStateProcessor) Start(redis redis.Client) error {
```

Comment:
Why do you have a separate Start function here? Why not make it part of the newTraceStateProcessor method? 

If it is valuable to keep it, maybe consider setting the client in the new... function and having this Start() being compatible with the startstopper model, in case we ever want to inject this.

Reply:
This probably is not a good name for this function and shouldn't have been exported. I changed it to `init`. 
I'm not putting it in `new` is because I don't want to `traceStateProcessor` to manage the `redis.Client`. I would like to keep the connection management in the `RedisBasicStore`